instructions/Add.cpp

```diff
@@ -1,426 +1,426 @@
+    set_flag(PARITY_OVERFLOW_FLAG, ((accumulator_ ^ result_16_) & ~(accumulator_ ^ data_8_)) & 0x80);
     // N is reset
     set_flag(ADD_SUB_FLAG, false);
     // C is set if carry from bit 7, else reset
-    set_flag(CARRY_FLAG, result_16 > 0x00FF);
+    set_flag(CARRY_FLAG, result_16_ > 0x00FF);
 
     // WZ register is updated using memory address
-    WZ_register = memory_address;
+    WZ_register_ = memory_address_;
 
-    accumulator = result_16;
+    accumulator_ = result_16_;
 
     //X & Y Flags are copies bit 3 & 5 of the accumulator
-    set_flag(X_FLAG, accumulator & 0x08);
-    set_flag(Y_FLAG, accumulator & 0x20);
+    set_flag(X_FLAG, accumulator_ & 0x08);
+    set_flag(Y_FLAG, accumulator_ & 0x20);
 }
 
 
 void z80cpu::ADD_implied_indexed_iy() {
-    t_state_cycles = 19;
+    t_state_cycles_ = 19;
 
-    displacement = static_cast<int8_t>(rom_read(program_counter));
-    program_counter++;
-    memory_address = index_register_y + static_cast<int16_t>(displacement);
-    data_8 = ram_read(memory_address);
-    result_16 = static_cast<uint16_t>(accumulator) + static_cast<uint16_t>(data_8);
+    displacement_ = static_cast<int8_t>(rom_read(program_counter_));
+    program_counter_++;
+    memory_address_ = index_register_y_ + static_cast<int16_t>(displacement_);
+    data_8_ = ram_read(memory_address_);
+    result_16_ = static_cast<uint16_t>(accumulator_) + static_cast<uint16_t>(data_8_);
 
     // S is set if result if negative, else reset
-    set_flag(SIGN_FLAG, result_16 & 0x0080);
+    set_flag(SIGN_FLAG, result_16_ & 0x0080);
     // Z is set if result is 0, else reset
-    set_flag(ZERO_FLAG, (result_16 & 0x00FF) == 0);
+    set_flag(ZERO_FLAG, (result_16_ & 0x00FF) == 0);
     // H is set if carry from bit 3, else reset
     // h_result = (accumulator & 0x0F) + (data & 0x0F)
-    set_flag(HALF_CARRY_FLAG, (accumulator & 0x0F) + (data_8 & 0x0F) > 0x0F);
+    set_flag(HALF_CARRY_FLAG, (accumulator_ & 0x0F) + (data_8_ & 0x0F) > 0x0F);
     // P/V is set if overflow, else reset
-    set_flag(PARITY_OVERFLOW_FLAG, ((accumulator ^ result_16) & ~(accumulator ^ data_8)) & 0x80);
+    set_flag(PARITY_OVERFLOW_FLAG, ((accumulator_ ^ result_16_) & ~(accumulator_ ^ data_8_)) & 0x80);
     // N is reset
     set_flag(ADD_SUB_FLAG, false);
     // C is set if carry from bit 7, else reset
-    set_flag(CARRY_FLAG, result_16 > 0x00FF);
+    set_flag(CARRY_FLAG, result_16_ > 0x00FF);
 
     // WZ register is updated using memory address
-    WZ_register = memory_address;
+    WZ_register_ = memory_address_;
 
-    accumulator = result_16;
+    accumulator_ = result_16_;
 
     //X & Y Flags are copies bit 3 & 5 of the accumulator
-    set_flag(X_FLAG, accumulator & 0x08);
-    set_flag(Y_FLAG, accumulator & 0x20);
+    set_flag(X_FLAG, accumulator_ & 0x08);
+    set_flag(Y_FLAG, accumulator_ & 0x20);
 }
 
 
 void z80cpu::ADD_implied_immediate() {
```

Comment:
if the ADD is it's own component vs the ADC maybe they can be nested objects with operations within the z80 class.